### PR TITLE
fix: fixing a console error for expired password view

### PR DESCRIPTION
**Pull request by anipendakur-okta**
OKTA-349016

## PR Checklist

- [X] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [X] Added e2e tests

- [X] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:
Before -
image

After -
image

### Reviewers:
@pradeepdewda-okta 

### Issue:

- [OKTA-349016](https://oktainc.atlassian.net/browse/OKTA-349016)

### CODE REVIEW

File: src/v2/view-builder/views/password/ReEnrollAuthenticatorPasswordView.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const Body = EnrollAuthenticatorPasswordView.prototype.Body.extend({
   },
 
   getPasswordPolicySettings () {
-    return this.options.appState.get('recoveryAuthenticator').settings;
+    return this.options.appState.get('recoveryAuthenticator')?.settings;
```
Comment (anipendakur-okta) [inline src/v2/view-builder/views/password/ReEnrollAuthenticatorPasswordView.js:20]: This is the only real change!

**Review by pradeepdewda-okta — APPROVED** [review]
🥇

File: src/v2/view-builder/views/password/ReEnrollAuthenticatorPasswordView.js [hunk 2]
```diff
@@ -17,7 +17,8 @@ const Body = EnrollAuthenticatorPasswordView.prototype.Body.extend({
   },
 
   getPasswordPolicySettings () {
-    return this.options.appState.get('recoveryAuthenticator').settings;
+    return this.options.appState.get('recoveryAuthenticator')?.settings
+      || this.options.appState.get('enrollmentAuthenticator')?.settings;
```
Comment (anipendakur-okta) [inline src/v2/view-builder/views/password/ReEnrollAuthenticatorPasswordView.js:21]: For more info, please refer to @SamSanjabi-okta 's comment here - https://oktainc.atlassian.net/browse/OKTA-349016?focusedCommentId=1641471

**Comment by pradeepdewda-okta** [comment]
@anipendakur-okta what's the screenshot when it uses `enrollmentAuthenticator` ?

**Comment by anipendakur-okta** [comment]
> @anipendakur-okta what's the screenshot when it uses `enrollmentAuthenticator` ?

It is seamless, ie, the password requirements will show up - 
image

Here is the corresponding JSON object -
`	"enrollmentAuthenticator": {
		"type": "object",
		"value": {
			"type": "password",
			"id": "aut8eqcFiHicejNIP0g4",
			"displayName": "Password",
			"methods": [{
				"type": "password"
			}],
			"settings": {
				"complexity": {
					"minLength": 8,
					"minLowerCase": 1,
					"minUpperCase": 1,
					"minNumber": 1,
					"minSymbol": 0,
					"excludeUsername": true,
					"excludeAttributes": []
				},
				"age": {
					"minAgeMinutes": 0,
					"historyCount": 0
				}
			}
		}
	},`

**Review by pradeepdewda-okta — APPROVED** [review]

**Pull request closed without merging**